app/policies/idv/flow_policy.rb

```diff
@@ -0,0 +1,158 @@
+module Idv
+  class FlowPolicy
+    include Rails.application.routes.url_helpers
+
+    NEXT_STEPS = Hash.new([])
+    NEXT_STEPS.merge!(
+      {
+        root: [:welcome],
+        welcome: [:agreement],
+        agreement: [:hybrid_handoff, :document_capture],
+        hybrid_handoff: [:link_sent, :document_capture],
+        link_sent: [:ssn],
+        document_capture: [:ssn], # in person?
+        ssn: [:verify_info],
+        verify_info: [:phone],
+        phone: [:phone_enter_otp],
+        phone_enter_otp: [:review],
+        review: [:personal_key],
+        personal_key: [:success],
+      },
+    )
+
+    attr_reader :idv_session, :user
+
+    def initialize(idv_session:, user:)
+      @idv_session = idv_session
+      @user = user
+    end
+
+    def step_allowed?(step:)
+      send(step)
+    end
+
+    def path_allowed?(path:)
+      step = path_to_step[path]
+      send(step)
+    end
```

Comment:
Seeing two `#send` calls is triggering my spidey sense. One of the issues with the old Flow State Machine was its reliance on dynamic dispatch via `#send`, which made tracing code and debugging a challenge.

I don't have a good alternative at the moment, but I'd love to get a sense of how we're handling this currently in the post-FSM world? There's not any red diff in this PR so it's hard to see what's being replaced and hard to get a sense of a pattern.

Reply:
Looking at how we have all the steps listed out in many places, maybe we could pull all the parts of this into one hash, basically defining a state machine.

We've be replacing the `#send` call with a hash lookup, but we'd be pulling a lot of separate repeated logic and keys together.

Just noodling here:

```ruby
STEPS = {
  root: Step.new(next_steps: [:welcome]),
  welcome: Step.new(
    path: idv_welcome_path,
    next_steps: [:agreement],
    requirements: ->(_user, _idv_session) { true },
  ),
  agreement: Step.new(
    path: idv_agreement_path,
    next_steps: [:hybrid_handoff, :document_capture],
    requirements: ->(_user, idv_session) { idv_session.welcome_visited },
  ),
  # ...
}
```